build/crds/generate-crd-docs.sh

```diff
@@ -0,0 +1,21 @@
+#! /usr/bin/env bash
+
+set -o errexit
+set -o pipefail
+
+SCRIPT_ROOT=$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)
+
+install_generator() {
+    go install github.com/ahmetb/gen-crd-api-reference-docs@latest
+}
+
+run_gen() {
+  install_generator
+  gen-crd-api-reference-docs \
+      -config="${SCRIPT_ROOT}/crd-docs-config.json" \
+      -template-dir="${SCRIPT_ROOT}/Documentation/gen-crd-api-reference-docs/template" \
+      -api-dir="github.com/rook/rook/pkg/apis/ceph.rook.io" \
```

Comment:
Can we also generate the docs for the OBCs? They are included in crds.yaml from the script `build-crds.sh`, but we don't have them in types.go, so perhaps this will be a challenge.

Reply:
Could you add more details regarding OBCs?, I think we can add them but of course we need some types.go to generate the docs, I can add the types I think, but need to have something similar like pkg/apis/ceph.rook.io/v1, to keep the same structure, do you have something in mind?

Reply:
The OBC types can be found in the [lib-bucket-provisioner repo](https://github.com/kube-object-storage/lib-bucket-provisioner/tree/master/pkg/apis/objectbucket.io/v1alpha1). Since COSI is coming soon, maybe we don't need to worry about this unless you find it simple to generate them now.

Reply:
I tried so add some types but I'm getting this error when I run `make codegen`, the error happens even without any change

```
$cd rook/
$make codegen 
Generating deepcopy funcs
F0427 14:39:05.518288  157208 main.go:82] Error: Failed executing generator: some packages had errors:
errors in package "github.com/rook/rook/pkg/apis/ceph.rook.io/v1":
open ../../../github.com/rook/rook/pkg/apis/ceph.rook.io/v1/zz_generated.deepcopy.go: no such file or directory
goroutine 1 [running]:
k8s.io/klog/v2.stacks(0x1)
	/home/javier/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:1026 +0x89
k8s.io/klog/v2.(*loggingT).output(0x986da0, 0x3, {0x0, 0x0}, 0xc020730000, {0x825437?, 0x1?}, 0xc020706820?, 0x0)
	/home/javier/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:975 +0x568
k8s.io/klog/v2.(*loggingT).printf(0x6?, 0x75fd40?, {0x0, 0x0}, {0x0, 0x0}, {0x73342a, 0x9}, {0xc020706820, 0x1, ...})
	/home/javier/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:750 +0x1d1
k8s.io/klog/v2.Fatalf(...)
	/home/javier/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:1502
main.main()
	/home/javier/src/rook/.cache/tools/linux_amd64/code-generator-0.20.0/cmd/deepcopy-gen/main.go:82 +0x49d

goroutine 6 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x0?)
	/home/javier/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:1169 +0x6a
created by k8s.io/klog/v2.init.0
	/home/javier/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:417 +0xf6
make: *** [Makefile:168: codegen] Error 255
```

Any idea why is it failing?
I tried to run it because we have a note in types.go file that we need to run generation each time that we update the types.go file

types.go
```
// ***************************************************************************
// IMPORTANT FOR CODE GENERATION
// If the types in this file are updated, you will need to run
// `make codegen` to generate the new types under the client/clientset folder.
// ***************************************************************************
```

Reply:
I'm not sure why that's failing, but it certainly looks like the codegen is outdated. When I ran the codegen locally I had a few differences and pushed them [here](https://github.com/travisn/rook/commit/a9669be55c4d2c360c490ff7b3cb0154419f1e41). Of course, those changes don't look related to your PR.